ecosystem/indexer-grpc/indexer-transaction-generator/src/config.rs

```diff
@@ -0,0 +1,136 @@
+        // Read the configuration file.
+        let config_raw = tokio::fs::read_to_string(&self.config)
+            .await
+            .with_context(|| format!("Failed to read configuration file: {:?}", self.config))?;
+
+        // Parse the configuration.
+        let config: TransactionGeneratorConfig = serde_yaml::from_str(&config_raw)
+            .with_context(|| format!("Failed to parse configuration file: {:?}", self.config))?;
+
+        // Run the transaction generator.
+        config.run(&self.output_folder).await
+    }
+}
+
+/// Overall configuration for the transaction generator.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TransactionGeneratorConfig {
+    pub import_config: TransactionImporterConfig, // TODO: Add scripted transaction generation configuration.
+}
+
+impl TransactionGeneratorConfig {
+    pub async fn run(&self, output_path: &Path) -> anyhow::Result<()> {
+        let import_config_path = output_path.join(IMPORTED_TRANSACTIONS_FOLDER);
+        // Check if the output folder exists.
+        if !import_config_path.exists() {
+            tokio::fs::create_dir_all(&import_config_path).await?;
+        }
+        self.import_config.run(&import_config_path).await
+    }
+}
+
+/// Configuration for importing transactions from multiple networks.
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct TransactionImporterConfig {
+    // Config is a map from network name to the configuration for that network.
+    #[serde(flatten)]
+    pub configs: HashMap<String, TransactionImporterPerNetworkConfig>,
+}
+
+impl TransactionImporterConfig {
+    pub async fn run(&self, output_path: &Path) -> anyhow::Result<()> {
+        // Validate the configuration. This is to make sure that no output file shares the same name.
+        let mut output_files = HashSet::new();
+        for (_, network_config) in self.configs.iter() {
+            for output_file in network_config.versions_to_import.values() {
+                if !output_files.insert(output_file) {
+                    return Err(anyhow::anyhow!(
+                        "[Transaction Importer] Output file name {} is duplicated",
+                        output_file
+                    ));
+                }
+            }
+        }
+        // Run the transaction importer for each network.
+        for (network_name, network_config) in self.configs.iter() {
+            network_config.run(output_path).await.context(format!(
+                "[Transaction Importer] Failed for network: {}",
+                network_name
+            ))?;
+        }
+        Ok(())
+    }
+}
+
+/// Configuration for importing transactions from a network.
+/// This includes the URL of the network, the API key, the version of the transaction to fetch,
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TransactionImporterPerNetworkConfig {
+    /// The endpoint of the transaction stream.
+    pub transaction_stream_endpoint: Url,
```

Comment:
should we have a default endpoint? since the url wouldn't really change. and this could also prevent from people putting wrong endpoint. For instance, I was able to run this without any issues. 
```import_config:
  testnet_config:
    # Transaction Stream endpoint addresss.
    transaction_stream_endpoint: https://grpc.testnet.aptoslabs.com:443
    # (Optional) The key to use with developers.aptoslabs.com
    api_key: aptoslabs_L7pzPsqiShp_EwzUCbhb6ehTe8PSvFp6ZeUr3B3gBdGaX
    # A map from versions to dump and their output names.
    versions_to_import:
      123: testnet_v1.json
  mainnet_config:
    # Transaction Stream endpoint addresss.
    transaction_stream_endpoint: https://grpc.testnet.aptoslabs.com:443
    # (Optional) The key to use with developers.aptoslabs.com
    api_key: aptoslabs_L7pzPsqiShp_EwzUCbhb6ehTe8PSvFp6ZeUr3B3gBdGaX
    # A map from versions to dump and their output names.
    versions_to_import:
      1234: testnet_v1234.json
      ```

Reply:
actually, if we have them already written in the config, people can just append more transactions to `versions_to_import`

Reply:
Let's enforce this at config side, not the code side. It's straightforward to give out a template with pre-populated endpoints.

Reply:
included a config template; let's start from here.